warehouse/integrations/snowflake/snowflake.go

```diff
@@ -756,56 +739,72 @@ func (sf *Snowflake) loadUserTables(ctx context.Context) map[string]error {
 			  ORDER BY
 				RECEIVED_AT DESC ROWS BETWEEN UNBOUNDED PRECEDING
 				AND UNBOUNDED FOLLOWING
-			) AS %[1]q
-`,
+			) AS %[1]q`,
 			colName,
 		)
 		firstValProps = append(firstValProps, firstValPropsQuery)
 	}
 
 	schemaIdentifier := sf.schemaIdentifier()
-	stagingTableName := warehouseutils.StagingTableName(provider, usersTable, tableNameLimit)
+	if sf.config.loadTableStrategy == loadTableStrategyAppendMode {
+		// doing an INSERT OVERWRITE because we used APPEND for the IDENTIFIES table as well and
+		// the COPY INTO command does not give us any reference as to what was added
+		sqlStatement := fmt.Sprintf(`
+			INSERT OVERWRITE INTO %[1]s.%[2]q ("ID", %[3]s)
+			SELECT DISTINCT *
+			FROM (
+				SELECT "USER_ID" as "ID", %[3]s
+				FROM %[1]s.%[4]q
+				WHERE "USER_ID" IS NOT NULL
+			);`,
+			schemaIdentifier, usersTable,
+			strings.Join(userColNames, ","),
+			identifiesTable,
+		)
```

Comment:
Let me know what you think about this approach and if you have some other ideas in mind. The problem is that when handling the `IDENTIFIES` table right before we're doing just a `COPY INTO` because we're in `APPEND` mode so we don't have any reference as to what it was copied and we can't therefore make a more selective `INSERT` here.

Anyway, since I'm under the impression that the users table is usually not that big beside some exceptions and considering that the `INSERT OVERWRITE INTO` is more efficient than a `COPY INTO` maybe we can give this a try.

Alternatively I was thinking that we could do two copies when in `APPEND` mode for the `IDENTIFIES`, one in `IDENTIFIES` and one in a `TEMPORARY TABLE`.

Reply:
The advantage of the `INSERT OVERWRITE INTO` is also that the `USERS` table will be duplicates free.

Reply:
@achettyiitr @lvrach :point_up:

Reply:
I'll check if I can benchmark this with an `IDENTIFIES` table that has 30 million records.

Reply:
For mattermost, one of the users table is having around 50M entries.

Reply:
Also, we would need to merge the traits before appending into users table.

Reply:
I was thinking we should probably do MERGE always for Users. Since if we keep on appending the entries, when we perform the merging of the traits, it's dataset will keep on increasing.
What do you think? @lvrach

Reply:
Yes, the merging of the traits complicates things since it adds a substantial amount of computation. I need to rethink think. Feedback as usual is welcome guys. We either always do the `MERGE` and or do 2 `COPY` for the `IDENTIFIES` so that here we can be more efficient with the `FIRST_VALUE`s.